src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -668,7 +708,9 @@ public void TestPriorityClassWindows()
         public void TestInvalidPriorityClass(ProcessPriorityClass priorityClass)
         {
             var process = new Process();
-            Assert.Throws<InvalidEnumArgumentException>(() => process.PriorityClass = priorityClass);
+            var exception = Record.Exception(() => process.PriorityClass = priorityClass);
```

Comment:
The `Assert.Throws<T>(Action)` method always failed as it said the type is an ArgumentException instead of an InvalidEnumArgumentException. It seems the type resolution is somewhat buggy. With the explicit type check `Assert.IsType<T>(object)` it succeeds.